### PR TITLE
Bump Rust toolchain to 2022-05-10

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

Our current toolchain does not have a miri component, thus, bumping the Rust version. As there is no point in updating to an old version, this bumps to the latest version.

## 🔍 What does this change?

Clippy got a bit more restrictive since our last toolchain. The lints were either applied or ignored depending on the situation

## 🚫  Blockers

- #551

### CODE REVIEW

**Comment by TimDiekmann** [comment]
Rebased onto #551 as CI needs to pass **all** crates.

File: packages/engine/lib/nano/src/lib.rs [hunk 1]
```diff
@@ -42,15 +41,6 @@
     clippy::missing_inline_in_public_items,
     reason = "We didn't any serious optimization work until now"
 )]
-#![allow(
```
Comment (Alfred-Mountfield) [inline packages/engine/lib/nano/src/lib.rs:45]: Are these all of the restriction lints or have we lost some?

**Review by Alfred-Mountfield — COMMENTED** [review]
One question otherwise looks good

File: packages/engine/lib/nano/src/lib.rs (hunk 1 above)
Comment (TimDiekmann) [reply packages/engine/lib/nano/src/lib.rs:45]: These were only allows for specific restriction lints.

Comment (TimDiekmann) [reply packages/engine/lib/nano/src/lib.rs:45]: Yes, right. I chose that restriction lints were way too restrictive (things like `pub use` is warned). Clippy suggests to only use specific lints from `restriction`, not the whole category.

Comment (Alfred-Mountfield) [reply packages/engine/lib/nano/src/lib.rs:45]: Alright, would have been good to get that reasoning 😄

**Review by Alfred-Mountfield — APPROVED** [review]

File: packages/engine/lib/nano/src/lib.rs (hunk 1 above)
Comment (TimDiekmann) [reply packages/engine/lib/nano/src/lib.rs:45]: Yes, sorry 🙃

**Pull request merged**